src/stream/src/executor/test_utils.rs

```diff
@@ -854,9 +854,6 @@ pub mod hash_join_executor {
             tx_r.push_chunk(chunk);
         }
 
-        tx_l.push_barrier(test_epoch(2), false);
-        tx_r.push_barrier(test_epoch(2), false);
```

Comment:
This second barrier will trigger a state store flush. Because the barrier before this is the initial barrier, it will not commit data to state store.